src/main/java/io/xpring/xrpl/model/XRPPayment.java

```diff
@@ -0,0 +1,117 @@
+package io.xpring.xrpl.model;
+
+import org.xrpl.rpc.v1.Payment;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.immutables.value.Value;
+import javax.annotation.Nullable;
+
+/**
+ * A payment on the XRP Ledger.
+ * @see "https://xrpl.org/payment.html"
+ */
+@Value.Immutable
+public interface XRPPayment {
+    static ImmutableXRPPayment.Builder builder() {
+        return ImmutableXRPPayment.builder();
+    }
+
+    /**
+     * @return The amount of currency to deliver.
+     */
+    XRPCurrencyAmount amount();
+
+    /**
+     * @return The unique address of the account receiving the payment.
+     */
+    String destination();
+
+    /**
+     * @return (Optional) Arbitrary tag that identifies the reason for the payment.
+     */
+    @Nullable Integer destinationTag();
+
+    /**
+     * @return (Optional) Minimum amount of destination currency this transaction should deliver.
+     */
+    @Nullable XRPCurrencyAmount deliverMin();
+
+    /**
+     * @return (Optional) Arbitrary 256-bit hash representing a specific reason or identifier for this payment.
+     */
+    @Nullable byte[] invoiceID();
+
+    /**
+     * @return (Optional) Array of payment paths to be used for this transaction.
+     * Must be omitted for XRP-to-XRP transactions.
+     */
+    @Nullable List<XRPPath> paths();
+
+    /**
+     * @return (Optional) Highest amount of source currency this transaction is allowed to cost.
+     */
+    @Nullable XRPCurrencyAmount sendMax();
+
+    static XRPPayment from(Payment payment) {
+        // amount is required
+        XRPCurrencyAmount amount = XRPCurrencyAmount.from(payment.getAmount().getValue());
+        if (amount == null) { return null; }
```

Comment:
You can shorten to 

```
if (!amount) { ... } 
```

Reply:
This is also giving me an issue about requiring a boolean when I try to write it like that...